packages/frontend/components/Sidebar/NUPathSection.tsx

```diff
@@ -0,0 +1,283 @@
+import { NUPathEnum, ScheduleCourse2 } from "@graduate/common";
+import { SidebarValidationStatus } from "./Sidebar";
+import { useState } from "react";
+import { Box, Flex, Spinner, Text } from "@chakra-ui/react";
+import { HelperToolTip } from "../Help";
+import {
+  CheckIcon,
+  ChevronDownIcon,
+  ChevronUpIcon,
+  SmallCloseIcon,
+} from "@chakra-ui/icons";
+
+interface NUPathSectionProps {
+  coursesTaken: ScheduleCourse2<unknown>[];
+  dndIdPrefix: string;
+  loading?: boolean;
+}
+const nuPathDisplayAndAbbr: [
+  nupath: string,
+  displayName: string,
+  abbreviation: string
+][] = [
+  [NUPathEnum.ND, "Natural and Designed World", "ND"],
+  [NUPathEnum.EI, "Creative Expression/Innovation", "EI"],
+  [NUPathEnum.IC, "Interpreting Culture", "IC"],
+  [NUPathEnum.FQ, "Formal and Quantitative Reasoning", "FQ"],
+  [NUPathEnum.SI, "Societies and Institutions", "SI"],
+  [NUPathEnum.AD, "Analyzing/Using Data", "AD"],
+  [NUPathEnum.DD, "Difference and Diversity", "DD"],
+  [NUPathEnum.ER, "Ethical Reasoning", "ER"],
+  [NUPathEnum.WF, "First Year Writing", "WF"],
+  [NUPathEnum.WD, "Advanced Writing in the Disciplines", "WD"],
+  [NUPathEnum.WI, "Writing Intensive", "WI"],
+  [NUPathEnum.EX, "Integration Experience", "EX"],
+  [NUPathEnum.CE, "Capstone Experience", "CE"],
+];
+const grey = "neutral.400";
+const green = "states.success.main";
+
+const NUPathSection: React.FC<NUPathSectionProps> = ({
+  coursesTaken,
+  dndIdPrefix,
+  loading,
+}) => {
+  const [opened, setOpened] = useState(false);
+
+  let validationStatus = SidebarValidationStatus.Error;
+
+  const nupathMap: Record<string, number> = {};
+
+  for (const course of coursesTaken) {
+    if (!course.nupaths) {
+      continue;
+    }
+
+    for (const nupath of course.nupaths) {
+      nupathMap[nupath] = (nupathMap[nupath] || 0) + 1;
+    }
+  }
+
+  const wiCount = nupathMap[NUPathEnum.WI];
```

Comment:
I would make it `const wiCount = nupathMap[NUPathEnum.WI] || 0;` that way you won't need to check for truthy-ness when checking if wiCount >= 2.

Reply:
big brain